course3/homework-vector/vector.cc

```diff
@@ -1,9 +1,55 @@
 #include <iostream>
 #include <cassert>
+#include <cmath>
 
 class Vector
 {
-    // TODO
+    private:
+        int private_x, private_y;
+
+    public:
+        // required constructors
+        Vector(int x, int y): private_x(x), private_y(y) {}
+        Vector() {
+            private_x = 0;
+            private_y = 0;
+        }
+
+        int GetX() { return private_x; }
+        int GetY() { return private_y; }
+
+        void SetX(int x) { private_x = x; }
+        void SetY(int y) { private_y = y; }
+
+        float GetTheta() {
+            return std::atan2(private_y, private_x) * 180 / M_PI;
+        }
+
+        float GetR() {
+            return std::pow(std::pow(private_x, 2) + std::pow(private_y, 2), 0.5);
+        }
+
+        friend std::ostream& operator<< (std::ostream& stream, Vector v) {
+            stream << "{(" << v.GetX() << ", " << v.GetY() << "), (" << v.GetR() << "@" << v.GetTheta() << ")}";
+            return stream;
+        }
+
+        bool operator== (Vector rhs) {
+            if (private_x == rhs.GetX() && private_y == rhs.GetY())
+                return true;
+            return false;
+        }
+
+        bool operator!= (Vector rhs) {
+            return !operator==(rhs);
+        }
+
+        Vector operator+ (Vector rhs) {
```

Comment:
I all the functions here, you take the parameter by value, which means deep-copying. In this case it's just 2 its just 2 ints, so no biggie, but for the sake of exercise - can you change the signatures so that the values are passed by reference instead of copy?